rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -55,252 +49,105 @@ class Airdrop(NamedTuple):
-        'YFI - Proof of Gucci Design Competition',
-    ),
-    'resuscitators': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/0ac0216f82f16453b74a40529384a152/raw/f003083090efd834bcee1d0d1fe4e218380aa0cf/poap_resucitators.json',
-        'https://poap.delivery/medalla-resuscitator',
-        'Medalla Resuscitators',
-    ),
-    'yam': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/d676e1f3db8df96928c2501c6be434ac/raw/fa0a603117f6e3e807a49491924aaaa2bc89179a/poap_yam.json',
-        'https://poap.delivery/yam',
-        'Yam Heroes',
-    ),
-    'ycover': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/727c05f3a9cab79059258528595c102e/raw/59ff367bd532f632ce94e69900709155a55de82a/poap_ycover.json',
-        'https://poap.delivery/ycover',
-        'A New Face For yCover',
-    ),
-    'yfi_og': (
-        # is checksummed
-        'https://gist.githubusercontent.com/LefterisJP/58862ec2b398c770d60c24e50e18e50c/raw/3292995f530baedcdacde02526c6b2bd1de0e110/poap_yfi_og.json',
-        'https://poap.delivery/yfi-og',
-        'I Played 4 YFI',
-    ),
-}
-
-
-def get_airdrop_data(name: str, data_dir: Path) -> Iterator[list[str]]:
-    airdrops_dir = data_dir / 'airdrops'
+def fetch_airdrops_metadata(database: 'DBHandler') -> tuple[dict[str, Airdrop], dict[str, list[str]]]:  # noqa: E501
+    """Fetches airdrop metadata from the rotki/data repository, parses them, and returns them in
+    two parts: a dict of Airdrop instances and a dict of POAP airdrops data.
+
+    May raise - RemoteError if the request fails or metadata is invalid.
+    """
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        airdrops_metadata_cached_at = globaldb_get_unique_cache_last_queried_ts_by_key(
+            cursor=cursor,
+            key_parts=(CacheType.AIRDROPS_METADATA,),
+        )
+
+    response = None
+    if ts_now() - airdrops_metadata_cached_at < HOUR_IN_SECONDS * 12:  # 12 hours
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            airdrops_metadata_cache = globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.AIRDROPS_METADATA,),
+            )
+        if airdrops_metadata_cache is not None:
+            response = jsonloads_dict(airdrops_metadata_cache)  # get cached response
+
+    if response is None:  # query fresh data
+        try:
+            response = requests.get(url=AIRDROPS_INDEX, timeout=CachedSettings().get_timeout_tuple()).json()  # noqa: E501
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Airdrops Index request failed due to {e!s}') from e
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.AIRDROPS_METADATA,),
+                value=rlk_jsondumps(response),
```

Comment:
Why use `rlk_jsondumps`? The data is normal json right? The `rlk_jsondumps` what does is to use `RKLEncoder` that has special logic for the data structures that we use in rotki

Reply:
The `response` here is a dict. So, getting `sqlite3.ProgrammingError: Error binding parameter 2: type 'dict' is not supported` if we don't dump it as a string